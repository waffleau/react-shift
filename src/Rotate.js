import React, {PropTypes, PureComponent} from 'react'

import Transition from './Transition'

export class Rotate extends PureComponent {
  static defaultProps = {
    from: 0
  }

  static propTypes = {
    ...Transition.baseProps,
    from: PropTypes.number,
    to: PropTypes.number.isRequired
  }

  getFromValue() {
    return {
      transform: `rotate(${this.props.from}deg)`
    }
  }

  getStyle() {
    const {style} = this.props
    return {
      transformOrigin: 'center center',
      ...style
    }
  }

  getToValue() {
    return {
      transform: `rotate(${this.props.to}deg)`
    }
  }

  render() {
    const {children} = this.props
    return (
      <Transition
        {...this.props}
        style={this.getStyle()}
        easing="easeInOut"
        fromValue={this.getFromValue()}
        toValue={this.getToValue()}>
        {children}
      </Transition>
    )
  }
}

export default Rotate
