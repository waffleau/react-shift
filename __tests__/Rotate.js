import React from 'react'
import renderer from 'react-test-renderer'

import Rotate from '../src/Rotate'

test('has no rotation when inactive', () => {
  const tree = renderer.create(
    <Rotate
      active={false}
      to={180}>
      {'Children'}
    </Rotate>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can rotate to 90 degrees', () => {
  const tree = renderer.create(
    <Rotate
      active
      to={90}>
      {'Children'}
    </Rotate>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can rotate to 180 degrees', () => {
  const tree = renderer.create(
    <Rotate
      active
      to={180}>
      {'Children'}
    </Rotate>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can rotate to 270 degrees', () => {
  const tree = renderer.create(
    <Rotate
      active
      to={270}>
      {'Children'}
    </Rotate>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
