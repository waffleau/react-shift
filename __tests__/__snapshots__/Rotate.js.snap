exports[`test can rotate to 90 degrees 1`] = `
<div
  style={
    Object {
      "transform": "rotate(90deg)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out 0ms",
    }
  }>
  Children
</div>
`;

exports[`test can rotate to 180 degrees 1`] = `
<div
  style={
    Object {
      "transform": "rotate(180deg)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out 0ms",
    }
  }>
  Children
</div>
`;

exports[`test can rotate to 270 degrees 1`] = `
<div
  style={
    Object {
      "transform": "rotate(270deg)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out 0ms",
    }
  }>
  Children
</div>
`;

exports[`test has no rotation when inactive 1`] = `
<div
  style={
    Object {
      "transform": "rotate(0deg)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out 0ms",
    }
  }>
  Children
</div>
`;
